library.py

```diff
@@ -63,3 +63,53 @@ def looking_for_a_dog(data: dict, name: str):
                 return user_dog_id
     except KeyError:
         return 'Такой собаки нет('
+
+
+def show_all_dogs(data: dict, data_base: dict) -> dict:
+    data['text'] = str(data_base)
+    send_message(data)
+    return data
+
+
+def add_new_dog(data: dict, data_base: dict) -> None:
+    new_dog = adding_a_dog(working_with_command(data['text']))
+    if new_dog != {}:
+        data_base[str(uuid4())] = new_dog
+        data['text'] = 'Собака успешно добавлена!'
+        send_message(data)
+        return
+    data['text'] = 'Не хватает аргументов'
+    send_message(data)
+
+
+def delete_dog(data: dict, data_base: dict) -> None:
+    info = working_with_command(data['text'])
+    name = info[1]
+    user_dog_id = looking_for_a_dog(data=data_base, name=name)
+    try:
+        if user_dog_id != 'Такой собаки нет(':
+            del data_base[user_dog_id]
+            data['text'] = 'Собака успешно удалена!'
+            send_message(data)
+    except KeyError:
+        data['text'] = 'Такой собаки нет('
+        send_message(data)
```

Comment:
Эксепшены надо ловить тогда, когда ты этого не можешь контролировать.
В данном случае ты можешь **легко** предотвратить его.
Например
```suggestion
if user_dog_id != 'Такой собаки нет(':
    if data_base.get(user_dog_id):
        del data_base[user_dog_id]
        data['text'] = 'Собака успешно удалена!'
        send_message(data)
    else:
        data['text'] = 'Такой собаки нет('
        send_message(data)

```